### DISCUSSION TITLE
API version management

### DISCUSSION THREAD

**Discussion by golu7679** [tokio-rs/axum#2802] [Q&A]
In axum, is their any feature to manage api version?

Like I want to create `/api/v1/login`, `/api/v2/login` multiple versions of api.

**Comment by dayvejones** [comment]
https://github.com/tokio-rs/axum/blob/main/examples/versioning/src/main.rs

You can also create several routers for versions 1 and 2 using the functions [nest](https://docs.rs/axum/latest/axum/struct.Router.html#method.nest) or [merge](https://docs.rs/axum/latest/axum/struct.Router.html#method.merge)

Maybe someone will come up with a more elegant solution 🙂

**Comment by calebkiage** [comment]
I created a sample that dispatches to handlers based on routes using semver here

https://github.com/calebkiage/axum-api-versioning-sample
